instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex

```diff
@@ -240,4 +242,14 @@ defmodule OpentelemetryEcto do
   defp add_additional_attributes(attributes, additional_attributes) do
     Map.merge(attributes, additional_attributes)
   end
+
+  defp telemetry_prefix(event_prefix) when is_list(event_prefix) do
+    event_prefix
+  end
+
+  defp telemetry_prefix(repo) when is_atom(repo) do
+    repo
+    |> Kernel.apply(:config, [])
+    |> Keyword.get(:telemetry_prefix)
```

Comment:
I think my main concern here would be if this will always return the right thing, or does it return `nil` in the case where a user hasn't overridden the default `telemetry_prefix` setting. I didn't look at the Ecto code, but the docs don't seem to indicate what fields will be there in the config - it's just whatever you configured in the app config or added in the `init/2` callback at runtime.

Reply:
What are the actionable items you would like me to take here?

Reply:
> What are the actionable items you would like me to take here?

I think he's asking how are you going to safeguard where the config doesn't have `telemetry_prefix` defined? This presumes it's there so if it isn't, it will return `nil`.

Reply:
Gathering notes over here,

Based on the docs, it seems that, at the very least, there is a default value: https://github.com/elixir-ecto/ecto/blob/c97c9bad699a0349d06877cef2ee0f12495eaee9/lib/ecto/repo.ex#L49-L57

Reply:
Based on https://github.com/elixir-ecto/ecto/blob/c97c9bad699a0349d06877cef2ee0f12495eaee9/lib/ecto/repo/supervisor.ex#L23-L28

There is still a possibility that somebody messes up the `repo_init`, so the question is, should we do defensive programming here or not?

I fixed the situation anyway